### PR TITLE
class for retail weeks

### PR DESCRIPTION
**Pull request by regatta-jlf**
* classes and supporting tests for retail weeks

### CODE REVIEW

File: src/RetailYear.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Regatta\Dates;
+
+/**
+ * A representation of a retail year (feb-jan).
+ */
+class RetailYear extends Range
```
Comment (duncan3dc) [inline src/RetailYear.php]: Isn't this identical to the `FinancialYear` class?

Comment (regatta-jlf) [inline src/RetailYear.php]: yes it is, shall I remove ?

Comment (duncan3dc) [inline src/RetailYear.php]: Yeh, things with different names that do the same thing is often confusing

Comment (regatta-jlf) [inline src/RetailYear.php]: ok, removed the year files now

**Comment by regatta-jlf** [comment]
generally a near copy of financial weeks date class , but with an additional test and of course week starting on a Sunday

File: src/Traits/PeriodFormatting.php [hunk 2]
```diff
@@ -92,4 +92,34 @@ public function getFinancialWeek()
...
+    {
+        # Get the start date of this financial year
+        $year = $this->getFinancialYear();
+        $start = Date::mkdate($year, 2, 1);
+
+        $range = new RangeInstance($start, $this);
+
+        $weekNumber = 0;
+
+        # If this year doesn't start on a monday then count the first few days as week 1
```
Comment (duncan3dc) [inline src/Traits/PeriodFormatting.php]: monday? or sunday?

File: src/Traits/PeriodFormatting.php [hunk 3]
```diff
@@ -92,4 +92,34 @@ public function getFinancialWeek()
...
+        if (!$start->isSunday()) {
+            ++$weekNumber;
+        }
+
+        # Then for each monday that starts a new week we increase our week number
```
Comment (duncan3dc) [inline src/Traits/PeriodFormatting.php]: monday? or sunday?

File: src/Traits/PeriodFormatting.php (hunk 2 above)
Comment (regatta-jlf) [inline src/Traits/PeriodFormatting.php]: thanks have amended

File: src/Traits/PeriodFormatting.php (hunk 3 above)
Comment (regatta-jlf) [inline src/Traits/PeriodFormatting.php]: thanks have amended

**Comment by duncan3dc** [comment]
Thanks :+1:

**Comment by regatta-jlf** [comment]
thanks, 
why did you merge it ? 
it is something to do with setup ?
I did not have chance to re base
will I be able to update my environments now 

Jeff

**Comment by duncan3dc** [comment]
The dates library is a little different than our others because it's open source. The Hive pull request I just pinged you on is what you need

**Comment by regatta-jlf** [comment]
thanks

**Pull request merged**